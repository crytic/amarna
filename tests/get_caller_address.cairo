func ERC721_transferFrom{pedersen_ptr: HashBuiltin*, syscall_ptr: felt*, range_check_ptr}(
    _from: felt, to: felt, token_id: Uint256
) {
    let (caller) = get_caller_address();
    let (is_approved) = _is_approved_or_owner(caller, token_id);
    assert is_approved = 1;

    _transfer(_from, to, token_id);
    return ();
}

func main() {
    ERC721_transferFrom();
    return ();
}
