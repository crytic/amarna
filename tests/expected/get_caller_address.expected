[must-check-caller-address] in get_caller_address.cairo:6:10
[unused-function] in get_caller_address.cairo:14:1